plugins/tedge_apt_plugin/src/lib.rs

```diff
@@ -103,29 +117,80 @@ fn run_op(apt: AptCli) -> Result<ExitStatus, InternalError> {
+                    Ok(filter) => Some(filter),
+                    Err(err) => {
+                        eprintln!("tedge-apt-plugin fails to list packages with matching name and maintainer: {err}");
+                        std::process::exit(1)
+                    }
+                },
+            };
+
+            let exclusive_filter = match (&exclusive_name, &exclusive_maintainer) {
+                (None, None) => None,
 
                 _ => match Regex::new(
                     format!(
-                        r"(^{}\t.*|^\S+\t\S+\t{}\s+.*)install ok installed",
-                        name.unwrap_or_default(),
-                        maintainer.unwrap_or_default()
+                        r"(^{}\t.*|^\S+\t\S+\t{}\s+.*)",
+                        exclusive_name.unwrap_or_default(),
+                        exclusive_maintainer.unwrap_or_default()
                     )
                     .as_str(),
                 ) {
-                    Ok(filter) => filter,
+                    Ok(filter) => Some(filter),
                     Err(err) => {
                         eprintln!("tedge-apt-plugin fails to list packages with matching name and maintainer: {err}");
                         std::process::exit(1)
                     }
                 },
             };
 
-            for line in stdout.trim_end().lines() {
-                if filter.is_match(line) {
-                    let (name, version) = get_name_and_version(line);
-                    println!("{name}\t{version}");
+            let installed_filter = Regex::new(r"install ok installed").unwrap();
+
+            match (inclusive_filter.as_ref(), exclusive_filter.as_ref()) {
+                (None, None) => {
+                    for line in stdout.trim_end().lines() {
+                        if installed_filter.is_match(line) {
+                            let (name, version) = get_name_and_version(line);
+                            println!("{name}\t{version}");
+                        }
+                    }
+                }
+                (Some(inclusive_filter), None) => {
+                    for line in stdout.trim_end().lines() {
+                        if installed_filter.is_match(line) && inclusive_filter.is_match(line) {
+                            let (name, version) = get_name_and_version(line);
+                            println!("{name}\t{version}");
+                        }
+                    }
+                }
+                (Some(inclusive_filter), Some(exclusive_filter)) => {
+                    for line in stdout.trim_end().lines() {
+                        if installed_filter.is_match(line)
+                            && (inclusive_filter.is_match(line) || !exclusive_filter.is_match(line))
+                        {
+                            let (name, version) = get_name_and_version(line);
+                            println!("{name}\t{version}");
+                        }
+                    }
```

Comment:
An alternative to avoid to repeat ourselves for the 4 cases (with and without include or exclude pattern), is to take as a default

- a pattern that always matches for the include regex (i.e. `".*"`)
- a pattern that never matches for the exclude regex (i.e. `"^$"`)

Reply:
> a pattern that always matches for the include regex (i.e. `".*"`)

I was wrong. This would defeat the purpose of the exclude filter when combined with the requirement that the include filter takes precedence over the exclude filter.